lib/common/output_text.c

```diff
@@ -336,6 +347,25 @@ pcmk__mk_text_output(char **argv) {
     return retval;
 }
 
+/*!
+ * \internal
+ * \brief Enable or disable fancy output for a text output object
+ *
+ * \param[in,out] out      Text output object
+ * \param[in]     enabled  Whether fancy output should be enabled for \p out
+ */
+void
+pcmk__output_text_set_fancy(pcmk__output_t *out, bool enabled)
+{
+    private_data_t *priv = NULL;
+
+    CRM_ASSERT((out != NULL)
+               && pcmk__str_eq(out->fmt_name, "text", pcmk__str_none));
```

Comment:
I could do that... Most/all of the existing format-specific functions like this use `CRM_CHECK()`. See `pcmk__output_xml_create_parent()`, `pcmk__output_xml_add_node_copy()`, `pcmk__output_create_xml_node()`, `pcmk__output_{get,set}_log_level()`, `pcmk__output_set_log_filter()`, etc. I thought I was following convention (I thought it was `CRM_ASSERT()`).

With that said, I also meant to create a text-specific output function for `option-list` that wraps the default function and sets fancy. I don't think there's ever a reason to use non-fancy output for `option-list`. It would be borderline-unreadable.

Reply:
I don't think check/assert is necessary to set a global behavior option, it's more like "if this is xml, do it this way". (The log set level/filter functions could fit that category too.) Functions that modify the XML in a specific way make sense to check/assert since they don't have a use case with any other type of output object.

Reply:
Makes sense